apps/user-office-backend/src/datasources/postgres/ProposalDataSource.ts

```diff
@@ -764,39 +766,42 @@ export default class PostgresProposalDataSource implements ProposalDataSource {
       throw new GraphQLError('Could not reset proposal events');
     }
 
-    const proposalEventsToReset: StatusChangingEventRecord[] = (
+    const proposalWorkflowId = proposalCall.proposal_workflow_id;
+
+    const proposalEventsToReset: (StatusChangingEventRecord &
+      ProposalWorkflowConnectionRecord)[] = (
       await database.raw(`
-          SELECT *
-          FROM proposal_workflow_connections AS pwc
-                   JOIN
-               status_changing_events
-               ON
-                       status_changing_events.proposal_workflow_connection_id = pwc.proposal_workflow_connection_id
-          WHERE pwc.proposal_workflow_connection_id >= (
-              SELECT proposal_workflow_connection_id
-              FROM proposal_workflow_connections
-              WHERE proposal_workflow_id = ${proposalCall.proposal_workflow_id}
-                AND proposal_status_id = ${statusId}
-          )
-            AND pwc.proposal_workflow_id = ${proposalCall.proposal_workflow_id};
+        SELECT *
+        FROM proposal_workflow_connections AS pwc
+        JOIN status_changing_events
+        ON status_changing_events.proposal_workflow_connection_id = pwc.proposal_workflow_connection_id
+        WHERE pwc.proposal_workflow_connection_id >= (
+          SELECT proposal_workflow_connection_id
+          FROM proposal_workflow_connections
+          WHERE proposal_workflow_id = ${proposalWorkflowId}
+          AND proposal_status_id = ${statusId}
+        )
+        AND pwc.proposal_workflow_id = ${proposalWorkflowId};
       `)
     ).rows;
 
     if (proposalEventsToReset?.length) {
-      const dataToUpdate = removeDuplicates(
-        proposalEventsToReset.map(
-          (event) =>
-            `${event.status_changing_event.toLocaleLowerCase()} = false`
-        )
-      ).join(', ');
+      const dataToUpdate: Record<string, boolean> = {};
+      proposalEventsToReset.forEach((event) => {
+        if (
+          !dataToUpdate.hasOwnProperty(
+            event.status_changing_event.toLocaleLowerCase()

```

Comment:
Suggestion: I think it would be nice to add some comments here, in which cases dataToUpdate would not have the property, or factor out the contents in condition into constant with descriptive name